### ISSUE TITLE
Phase Detection Learning 

### ISSUE THREAD

**Issue by Bryanrt-geophys** [smousavi05/EQTransformer#82]
Hello Mustafa,

I believe I have been been having success so far with your work. I completed my pipeline code (if you can call it that) that draws from unstructured label databases to put together a STEAD formatted set to pass to EqT. Thus far I have put together a training set with ~25k events and I am starting to see some results, but I am curious about what is normal for expectations in when to see learning of phase detection. 

My F scores where zero's for detections and phases from 100 examples to 1,000 examples and then I suddenly had detection and val_detection jump up in score and start to logarithmically converge on a score of 0.8. My F scores have maintained this behavior from 1k events out to ~25k events:

![X_learning_curve_f1](https://user-images.githubusercontent.com/76978628/126910818-cc049aae-d4ae-41c6-9258-a1a627fbdabc.png) 

![X_learning_curve_loss](https://user-images.githubusercontent.com/76978628/126910827-44b31407-ee7b-4e0d-83c3-ea92359f25dd.png)

I am curious if my phase detections may suddenly show up in F scores if I keep appending my training set, or if there is reason to believe there may be an issue with my data. Any input?

When I use the `tester()`, the output figures are detecting events and P-phases but the S-phase is very, very week. 

![DAG SC_20201005012800_EV](https://user-images.githubusercontent.com/76978628/126910893-f16be45f-4d8c-47a0-87c4-e228b0f7ad08.png)

![PECS TX_20210623141700_EV](https://user-images.githubusercontent.com/76978628/126910882-7796baf7-8f0d-47e1-8a52-f42626b7dc35.png)
 
I have been looking at my input values and noticed there are a number of Nan values in my `s_weight` attr. I don't think this is an issue though, correct?

I just don't want to put a lot of energy into appending my data set if there is any obvious issue with my data that needs to be fixed first. Thanks for all the help!

Bryan

**Comment by Bryanrt-geophys** [comment]
Oh, also, I have been thinking about merging my data set with the STEAD data for a transfered learning approach by turning off the shuffle feature and manually splitting the training/dev/test sets so all of my data is dispersed between dev/test. Any thoughts?

## Data segmentation for transfer learning
training(100%STEAD)/dev(STEAD shuffled with 50% NMTSO)/test(STEAD shuffled with 50% NMTSO)

**Comment by smousavi05** [comment]
@Bryanrt-geophys Hi Bryan, these are looking good and encouraging.
1) Nan values are not good. TF doesn't like them try to replace them with zeros. 
2) From your test plots I can see that the model has started to learn a little bit of P picking. You just need to continue the training for more epoches > 100-200. You have a relatively smaller data set with relatively noisy waveforms. So it is harder to train the model. Just keep training for longer time. You can ease the early stopping criteria to let the network keep training. 
3) having all your data in dev or test set is not a good idea. Either try to fine tune a pre-trained EqT model using your data (70% for training and the rest for dev/test). Or just append them to STEAD and then randomly select 85% for training 5% dev an the rest for testing.

**Comment by Bryanrt-geophys** [comment]
Awesome, thanks for the great advice on all points. I've been running with a mini-batch of 200 and 20 epochs. I'll set the epochs out farther. 

I'm regards to the Nan's, would you recommend changing my workflow to convert those to `None` or make an attempt to extrapolate them?

Thanks for the tip to use Obspy directly to try to get all channels in the mseed files. I'll give that a shot immediately.

Best,
Bryan

**Comment by smousavi05** [comment]
I would replace Nan's

**Comment by Bryanrt-geophys** [comment]
Hello again @smousavi05, 

Does EqT require the signal in the hdf file to be centered on 1 and normalized to avoid passing negative values to the ReLU function or is that a built in processing step in `trainer`?

I currently have all of my training data shifted so they are centered on zero.
